async_service/nats_pub_sub.py

```diff
@@ -135,6 +153,23 @@ async def publish_output_message(
             timeout=5,
         )
 
+    async def get_output_message(self, request_id: str, timeout: float = 0.5) -> bytes:
+        if timeout > 2:
+            raise ValueError("Timeout must be less than 2 seconds")
```

Comment:
Any strong reasons?

Reply:
this has been kept low to ensure that a lot of these calls do not block the server

Reply:
> this has been kept low to ensure that a lot of these calls do not block the server

This should not block the server, given that this is already implemented in the "async" fashion.

Reply:
Also, let us not add this validation here. Move it in the server code if the server is worried about getting blocked. Otherwise, you need to implement is for other "result stores" too.